include/dlaf/permutations/general/impl.h

```diff
@@ -152,4 +156,86 @@ void Permutations<B, D, T, C>::call(SizeType i_begin, SizeType i_end, Matrix<con
                                                false>(dlaf::internal::Policy<B>(), std::move(permute_fn),
                                                       std::move(sender)));
 }
+
+template <class T, Device D, Coord C, class OfssetsArrSender>
+void gatherData(const comm::Communicator& comm, comm::IndexT_MPI root_rank,
+                OfssetsArrSender&& offsets_sender, Matrix<T, D>& send_mat, Matrix<T, D>& recv_mat) {
+  //(void) offssets_sender;
+
+  // Note: the leading dimension is equal to the the size of the region.
+  SizeType lenvecs;  // TODO: calculate subproblem size
+  auto gather_f = [comm, root_rank, lenvecs](const std::vector<SizeType>& offsets,
```

Comment:
Is this something that could live in `dlaf/communication/kernels.h` (in its current, or a more abstracted form...)?